sparse_ho/algo/implicit.py

```diff
@@ -61,35 +60,13 @@ def get_beta_jac_t_v_implicit(
     if sol_lin_sys is not None and not hasattr(model, 'dual'):
         sol0 = init_dbeta0_new(sol_lin_sys, mask, mask0)
     else:
-        size_mat = mat_to_inv.shape[0]
-        sol0 = np.zeros(size_mat)
-    try:
-        sol = cg(
-            mat_to_inv, - model.generalized_supp(X, v, log_alpha),
-            # x0=sol0, tol=tol, maxiter=1e5)
-            x0=sol0, tol=tol)
-        if sol[1] == 0:
-            sol_lin_sys = sol[0]
-        else:
-            raise ValueError('cg did not converge.')
-    except Exception:
-        print("Matrix to invert was badly conditioned")
-        size_mat = mat_to_inv.shape[0]
-        if is_sparse:
-            reg_amount = 1e-7 * norm(model.reduce_X(X, mask).todense(),
-                                     ord=2) ** 2
-            mat_to_inv += reg_amount * identity(size_mat)
-        else:
-            reg_amount = 1e-7 * norm(model.reduce_X(X, mask), ord=2) ** 2
-            mat_to_inv += reg_amount * np.eye(size_mat)
-        sol = cg(
-            mat_to_inv + reg_amount * identity(size_mat),
-            - model.generalized_supp(X, v, log_alpha),
-            x0=sol0, atol=1e-3)
+        sol0 = None  # TODO add warm start for SVM and SVR
+    sol = cg(
+        mat_to_inv, - model.generalized_supp(X, v, log_alpha),
+        x0=sol0, tol=tol_lin_sys, maxiter=max_iter_lin_sys)
 
     sol_lin_sys = sol[0]
 
     jac_t_v = model._get_jac_t_v(
```

Comment:
honest question: why do we need to pass a copy of v ? having a function modify its output is against the principle of least astonishment for me, here

Reply:
done

Reply:
I still see v.copy() below